nixos/modules/services/games/pterodactyl-wings.nix

```diff
@@ -0,0 +1,214 @@
+            };
+            port = mkOption {
+              type = port;
+              default = 8080;
+              description = "The port that the internal webserver should bind to.";
+            };
+            ssl = {
+              enabled = mkEnableOption "HTTPS for the API interface";
+              cert = mkOption {
+                type = nullOr path;
+                default = null;
+                description = "Path to certificate to use for HTTPS.";
+              };
+              key = mkOption {
+                type = nullOr path;
+                default = null;
+                description = "Path to key to use for HTTPS.";
+              };
+            };
+          };
+          system = {
+            root_directory = mkOption {
+              type = path;
+              default = "/var/lib/pterodactyl";
+              description = "Root directory where daemon data is stored.";
+            };
+            log_directory = mkOption {
+              type = path;
+              default = "/var/log/pterodactyl";
+              description = "Directory where logs for server installations and other wings events are logged.";
+            };
+            data = mkOption {
+              type = path;
+              defaultText = "/var/lib/pterodactyl/volumes";
+              description = "Directory where game server data is stored.";
+            };
+            archive_directory = mkOption {
+              type = path;
+              defaultText = "/var/lib/pterodactyl/archives";
+              description = "Directory where server archives for cross-node transferring will be stored.";
+            };
+            backup_directory = mkOption {
+              type = path;
+              defaultText = "/var/lib/pterodactyl/backups";
+              description = "Directory where local backups will be stored.";
+            };
+            username = mkOption {
+              type = str;
+              default = "pterodactyl";
+              description = ''
+                The user that should own all of the server files, and be used for containers.
+                NixOS will create this user and group if left at the default.
+              '';
+            };
+            sftp = {
+              bind_address = mkOption {
+                type = str;
+                default = "0.0.0.0";
+                example = "[::1]";
+                description = "The bind address of the SFTP server.";
+              };
+              bind_port = mkOption {
+                type = port;
+                default = 8080;
+                description = "The bind port of the SFTP server.";
+              };
+              read_only = mkOption {
+                type = bool;
+                default = false;
+                description = "If set to true, no write actions will be allowed on the SFTP server.";
+              };
+            };
+          };
+          docker = {
+            network = {
```

Comment:
Should we assert this with the options for docker/podman?

Reply:
Not sure how to handle this with services that don't use the nixos options to manage their networks (like this one), and even then which nixos options would show the locally assigned options.